### ISSUE TITLE
Documentation Overhaul 

### ISSUE THREAD

**Issue by robellegate** [IdentityPython/SATOSA#271]
The documentation for this project needs an overhaul. There isn't a lot of structure to the current documentation, it's very difficult to find how to configure the various frontend/backend plugins without either digging into the codebase to see how the configuration files are being serialized.

For long-term support, I believe this project needs better documentation. It should be using either something like pydoc, which is already being used by the [pysaml2](../../pysaml2) project, or something like MkDocs.

I am willing to help contribute to efforts of improving documentation, but I wanted to leave the decision of if/how it's implemented to the maintainers and community.
## Code Version

**Branch:** Master
**Commit:** d8bb07a4e82494623529bfae01bbb4dcab714585

**Comment by peppelinux** [comment]
A readthedocs page would be a good Place I think.

Probably we should also document all the pysaml2 configuration options that
are actually undocumented at this moment. All the Saml2 backend/frontend
parameters rely on those belonging to pysaml2.

Il gio 1 ago 2019, 16:47 Robert Ellegate <notifications@github.com> ha
scritto:

**Comment by robellegate** [comment]
> Probably we should also document all the pysaml2 configuration options that are actually undocumented at this moment. All the Saml2 backend/frontend parameters rely on those belonging to pysaml2.

I completely agree. I didn't know the local metadata provider type could work with both aggregate files and directories until I dug into the [pysaml2 source code](https://github.com/IdentityPython/pysaml2/blob/master/src/saml2/mdstore.py#L887-L897).

**Comment by peppelinux** [comment]
Also related to
https://github.com/IdentityPython/SATOSA/issues/217

**Comment by FabianGrob** [comment]
Hi,
I am trying to use Satosa to implement a solution for several idps, and i do not find it completely clear how idps are added to satosa configuration.
Maybe it is a language barrier, can sameone give a detailed guide on how to add new idps. I found some as google in the documentation but i cannot reproduce them.

Thanks in advance

**Comment by vladimir-mencl-eresearch** [comment]
> Hi, I am trying to use Satosa to implement a solution for several idps, and i do not find it completely clear how idps are added to satosa configuration. Maybe it is a language barrier, can sameone give a detailed guide on how to add new idps. I found some as google in the documentation but i cannot reproduce them.
> 
> Thanks in advance


Hi,

This is not really place to ask documentation questions ... but, if you are after configuring multiple IdPs for a SAML SP (using the saml2 backend), you can just add more entries into the `metadata` part of the config - as here in the sample config: https://github.com/IdentityPython/SATOSA/blob/master/example/plugins/backends/saml2_backend.yaml.example#L31

Cheers,
Vlad

**Comment by peppelinux** [comment]
Hi,

Try the readme here
https://github.com/italia/Satosa-Saml2Spid

best


Il mar 5 lug 2022, 19:47 FabianGrob ***@***.***> ha scritto:

**Comment by FabianGrob** [comment]
> 
Where may i ask this kind of questions?

**Comment by vladimir-mencl-eresearch** [comment]
> > 
> 
> Where may i ask this kind of questions?

The bottom of the README file has:
> If you have any questions regarding operations/deployment of SATOSA please use the satosa-users [mailing list](https://lists.sunet.se/postorius/lists/idpy-discuss.lists.sunet.se/).

So https://lists.sunet.se/postorius/lists/idpy-discuss.lists.sunet.se/
